### PR TITLE
Add cert monitor manifest

### PR DESCRIPTION
**Pull request by rikatz**
Cert-monitor will read the metrics from cert-manager prometheus exporter, transform it into days and push to stackdriver.

TODO:
* Turn the GCP Project ID configurable
* Improve the Prometheus and GKE exporter security to drop privileges

### CODE REVIEW

File: cert-manager/cert-monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        volumeMounts:
+        - mountPath: /etc/prometheus
+          name: exporterconfig
+        - mountPath: /data
+          name: promdata
+      - name: sidecar
+        image: gcr.io/stackdriver-prometheus/stackdriver-prometheus-sidecar:0.8.2
+        imagePullPolicy: Always
+        args:
+        - --stackdriver.project-id=fluent-plate-304118
```
Comment (ameukam) [inline cert-manager/cert-monitor.yaml:65]: Do you mean `kubernetes-public` ?

File: cert-manager/cert-monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        - --stackdriver.generic.location=us-central1-c
```
Comment (ameukam) [inline cert-manager/cert-monitor.yaml:66]: Can we use `us-central` instead ?

**Comment by ameukam** [comment]
/assign @spiffxp @thockin 
/hold
If others people want to comment

File: cert-manager/cert-monitor.yaml (hunk 1 above)
Comment (rikatz) [reply cert-manager/cert-monitor.yaml:65]: @ameukam thanks! Yes, it needs to be the project id, so I'll change here.

File: cert-manager/cert-monitor.yaml (hunk 2 above)
Comment (rikatz) [reply cert-manager/cert-monitor.yaml:66]: Let me test and I'll let you know :)

Comment (rikatz) [reply cert-manager/cert-monitor.yaml]: Corrected, it works fine :)

File: cert-manager/cert-monitor.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      rules:
+      - record: certificate_expire_remaining_days 
+        expr: (certmanager_certificate_expiration_timestamp_seconds - time())/86400
+  sdexporter.yaml: |
+    static_metadata:
+    - metric: certificate_expire_remaining_days
+      type: gauge 
+      value_type: double 
+      help: Seconds until a certificate days
+kind: ConfigMap
```
Comment (thockin) [inline cert-manager/cert-monitor.yaml:28]: If humans are going to manage this, maybe move all these to the top of each block? :)

File: cert-manager/cert-monitor.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prom-certmanager
+  template:
+    metadata:
+      labels:
+        app: prom-certmanager
+    spec:
+      containers:
```
Comment (thockin) [inline cert-manager/cert-monitor.yaml:50]: You want to specify resources for these, right?

File: cert-manager/cert-monitor.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      type: gauge 
+      value_type: double 
+      help: Seconds until a certificate days
+kind: ConfigMap
+metadata:
+  name: exporterconfig
+  namespace: cert-manager
+---
+apiVersion: apps/v1
+kind: Deployment
```
Comment (thockin) [inline cert-manager/cert-monitor.yaml:34]: I would love some comments (yay YAML, you beat JSON at something) to explain what's going on.  Or a README-ish file?

File: cert-manager/cert-monitor.yaml (hunk 3 above)
Comment (rikatz) [reply cert-manager/cert-monitor.yaml:28]: what if those are humans that deal with lisp yamls, and read from the end to the beginning? :P

Kidding, corrected here!

File: cert-manager/cert-monitor.yaml (hunk 4 above)
Comment (rikatz) [reply cert-manager/cert-monitor.yaml:50]: Yes! Also I do want to make this image run as non-root, just didn't had enough time to test it, but I'll test right now and just push the manifest corrected

File: cert-manager/cert-monitor.yaml (hunk 5 above)
Comment (rikatz) [reply cert-manager/cert-monitor.yaml:34]: I'll make both :) 

Put some comments on the YAML, at least about the prometheus and sdexporter configuration, and write on the README (I've already added a README on the another PR)

**Comment by rikatz** [comment]
@thockin added some limits, some comments, some security context :)

Tested in my cluster and worked fine, hopefully this is the first step to renew the certificates before anything blows.

Next step: I'll check a way to automate the dashboard and alerting (there's an issue assigned to me anyway) so we can somehow automate everything.

**Comment by spiffxp** [comment]
Checking in... do we still need this?  I admit I have kinda lost track at where we are with using cert-manager vs. GKE managed certificates these days

**Comment by rikatz** [comment]
/close

As we discussed, per migration to Google managed certificates this is not necessary anymore :)

**Pull request closed without merging**